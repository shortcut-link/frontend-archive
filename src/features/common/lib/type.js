//@flow
export type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';

export type Options = {
  headers?: { [key: string]: string },
  parse?: 'text' | 'json' | 'noparse',
  baseURI?: string,
  body?: string | FormData | mixed
};

export type ResponseOk<R> = {
  ok: true,
  result: R
};

export type ResponseError<E> = {
  ok: false,
  error: E
};

export type CustomResponse<R, E> = ResponseOk<R> | ResponseError<E>;
