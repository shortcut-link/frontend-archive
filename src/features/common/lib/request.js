//@flow
import { $token } from '../model/token';
import type { Method, Options, CustomResponse } from './type';

const baseURI = '/api';

export const request = (
  method: Method,
  url: string,
  options: Options = {}
): Promise<any> => {
  const token = $token.getState();

  const headers = new Headers({
    ...createContentType(options),
    ...createAuthorization(token),
    ...options.headers
  });

  const uri = `${options.baseURI || baseURI}${url}`;

  const { body, ...restOptions } = options;

  const config = new Request(uri, {
    method,
    headers,
    ...restOptions,
    body: createBody(options, headers)
  });

  return fetch(config).then(response => {
    if (options.parse === 'text') {
      return response.text;
    }

    if (options.parse === 'noparse') {
      return response.text;
    }

    const contentType = response.headers.get('Content-Type');

    if (contentType && contentType.includes('json')) {
      return response.json().then(responseToPromise, responseToPromise);
    }

    throw new TypeError('Unexpected content-type');
  });
};

const createContentType = (options: Options) => {
  const header = contentTypeFromOptions(options);

  return header ? { 'Content-Type': header } : {};
};

const contentTypeFromOptions = (options: Options) => {
  if (options && options.headers && options.headers['Content-Type']) {
    return options.headers['Content-Type'];
  }

  if (options && options.body && options.body instanceof FormData) {
    return 'multipart/form-data';
  }

  return typeof options.body === 'object' ? 'application/json' : '';
};

const createAuthorization = (token: ?string) =>
  token ? { Authorization: `bearer ${token}` } : {};

const createBody = (
  options: Options,
  headers: Headers
): FormData | string | void => {
  const contentType = headers.get('content-type');

  if (options.body && contentType && contentType.includes('json')) {
    return JSON.stringify(options.body);
  }

  if (options.body instanceof FormData) {
    return options.body;
  }

  return undefined;
};

function responseToPromise<R, E>(response: CustomResponse<R, E>) {
  return response && typeof response.ok === 'boolean'
    ? okToPromise<R, E>(response)
    : response;
}

function okToPromise<R, E>(response: CustomResponse<R, E>): Promise<R> {
  return response.ok
    ? Promise.resolve(response.result)
    : Promise.reject(response.error);
}
