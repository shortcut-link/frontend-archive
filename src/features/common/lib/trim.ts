import { SyntheticEvent } from 'react';

export const trimEvent = (event: SyntheticEvent<HTMLInputElement>) =>
  event.currentTarget.value.trim();
