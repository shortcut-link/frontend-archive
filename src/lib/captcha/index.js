import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';

const key = '6LdPb7MUAAAAAEEVmEYPQq9kE6TzPPgfcBwua57l';

/**
 * reCaptcha
 * @param {{ onChange: function }} onChange Callback that is called when passing a captcha
 */
export const Captcha = ({ onChange }) => (
  <ReCAPTCHA sitekey={key} onChange={onChange} />
);
