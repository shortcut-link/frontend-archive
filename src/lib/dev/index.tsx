import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter } from 'react-router-dom';

import { lightTheme } from 'ui/themes';

/**
 * For testing components that require ThemeProvider or BrowserRouter
 */
export const ThemeRouterProvider = (component: React.ReactNode) =>
  renderer.create(
    <ThemeProvider theme={lightTheme}>
      <BrowserRouter>{component}</BrowserRouter>
    </ThemeProvider>
  );
