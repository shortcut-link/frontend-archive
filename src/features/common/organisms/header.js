import React from 'react';
import styled from 'styled-components';

import { Container } from 'ui/templates';
import { WithThemeToggler } from 'lib/theme';
import { ToggleSelectTheme } from 'ui/molecules';
import { Authenticated } from './authenticated';
import { Link } from 'ui/atom';
import { pathRoutes } from 'features/join';

export const Header = () => (
  <ContainerBox>
    <Container className={'container'}>
      <Logo />
      <RightPanel>
        <Account />
        <ToggleThemeButton />
      </RightPanel>
    </Container>
  </ContainerBox>
);

const Logo = () => {
  return (
    <Link to={'/'} style={{ textTransform: 'uppercase' }}>
      Shortcut-link
    </Link>
  );
};

const Account = () => (
  <Authenticated
    renderExists={() => TextAccount(true)}
    renderEmpty={() => TextAccount(false)}
  />
);

const TextAccount = auth => {
  const link = auth ? '/profile' : pathRoutes['join'];
  const text = auth ? 'Ваш аккаунт' : 'Войти в аккаунт';

  return (
    <Link to={link} style={{ marginRight: '20px' }}>
      {text}
    </Link>
  );
};

const ToggleThemeButton = () => (
  <WithThemeToggler
    render={({ isDark, toggle }) => {
      return <ToggleSelectTheme toggle={toggle} isDark={isDark} />;
    }}
  />
);

const RightPanel = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ContainerBox = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 5rem;

  & > div.container {
    justify-content: space-between;
    align-items: center;
  }

  ${({ theme }) => theme.embed.canvas}
`;
