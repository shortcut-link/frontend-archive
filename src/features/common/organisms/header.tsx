import React from 'react';
import styled from 'styled-components';

import { Container } from 'ui/templates';
import { WithThemeToggler } from 'lib/theme';
import { ToggleSelectTheme } from 'ui/molecules';
import { Authenticated } from './authenticated';
import { Link } from 'ui/atom';
import { routesPath } from 'pages';

export const Header = () => (
  <ContainerBox>
    <Container className={'container'}>
      <Logo />
      <RightPanel>
        <Account />
        <ToggleThemeButton />
      </RightPanel>
    </Container>
  </ContainerBox>
);

const Logo = () => {
  return (
    <h1>
      <Link
        to={routesPath.home}
        style={{
          textTransform: 'uppercase'
        }}
      >
        Shortcut-link
      </Link>
    </h1>
  );
};

const Account = () => {
  const AccountEmpty = () => (
    <Link to={routesPath.join.login}>Login to account</Link>
  );
  const AccountExists = () => <Link to={routesPath.profile}>Your account</Link>;

  return (
    <Authenticated renderExists={AccountExists} renderEmpty={AccountEmpty} />
  );
};

const ToggleThemeButton = () => (
  <WithThemeToggler
    render={({ isDark, toggle }) => (
      <ToggleSelectTheme switching={() => toggle()} isDark={isDark} />
    )}
  />
);

const RightPanel = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  & > :not(:first-child) {
    margin-left: 20px;
  }
`;

const ContainerBox = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 5rem;

  & > div.container {
    justify-content: space-between;
    align-items: center;
  }

  ${({ theme }) => theme.embed.canvas}
`;
