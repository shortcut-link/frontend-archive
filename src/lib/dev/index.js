import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { ToggleThemeProvider } from 'lib/theme';
import { lightTheme, darkTheme } from 'ui/themes';

/* For testing components that
 * require ThemeProvider or BrowserRouter
 */

export const ThemeProviderRouter = ({ children }) => (
  <ToggleThemeProvider light={lightTheme} dark={darkTheme}>
    <BrowserRouter>{children}</BrowserRouter>
  </ToggleThemeProvider>
);
