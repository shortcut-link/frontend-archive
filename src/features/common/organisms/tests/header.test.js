import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from '../header';
import { ThemeProviderRouter } from 'lib/dev';

it('must render Header', () => {
  const component = (
    <ThemeProviderRouter>
      <Header />
    </ThemeProviderRouter>
  );

  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
