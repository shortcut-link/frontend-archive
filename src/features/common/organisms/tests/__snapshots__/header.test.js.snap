// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must render Header 1`] = `
.c1 {
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  width: initial;
  height: initial;
  min-width: initial;
  max-width: initial;
  min-height: initial;
  max-height: initial;
  padding: initial;
  display: inherit;
  max-width: 50rem;
  width: 100%;
  padding: 1rem;
}

.c4 {
  position: relative;
  display: block;
  width: calc((1.4rem * 2) + 8px);
  height: calc(1.4rem + 8px);
}

.c8 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: var(--border);
  border-radius: 1.4rem;
  cursor: pointer;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  background-color: var(--primary);
}

.c8:before {
  content: "";
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 4px;
  left: 4px;
  width: 1.4rem;
  height: 1.4rem;
  font-size: calc(1.4rem / 2);
  line-height: 0px;
  background-color: white;
  border-radius: 50%;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c8:before {
  content: '☀️';
}

.c5 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c5:checked {
  -webkit-transform: translateX(1.4rem);
  -ms-transform: translateX(1.4rem);
  transform: translateX(1.4rem);
}

.c5:checked + .c6 {
  background-color: var(--primary);
}

.c5:checked + .c6::before {
  -webkit-transform: translateX(1.4rem);
  -ms-transform: translateX(1.4rem);
  transform: translateX(1.4rem);
}

.c5:checked + .c7::before {
  content: '🌙';
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  color: inherit;
  font-size: inherit;
  outline: none;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  width: initial;
  height: initial;
  min-width: initial;
  max-width: initial;
  min-height: initial;
  max-height: initial;
  padding: initial;
}

.c2:hover,
.c2:focus {
  color: var(--primary-hover);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > :not(:first-child) {
  margin-left: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5rem;
  color: var(--canvas-text);
  background-color: var(--canvas);
}

.c0 > div.container {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  <div
    className="container c1"
  >
    <h1>
      <a
        className="c2"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        Shortcut-link
      </a>
    </h1>
    <div
      className="c3"
    >
      <a
        className="c2"
        href="/join"
        onClick={[Function]}
      >
        Login to account
      </a>
      <label
        className="c4"
        onKeyPress={[Function]}
        radius="1.4rem"
      >
        <input
          checked={false}
          className="c5"
          onChange={[Function]}
          radius="1.4rem"
          type="checkbox"
        />
        <span
          className="c6 slider c7 c8"
          radius="1.4rem"
        />
      </label>
    </div>
  </div>
</div>
`;
